File: 0001_0100/0071/0071.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    p = [int(x) for x in f.readline().split()]
    s = sum(p)
    for i in range(1, 1 << (n - 1)):
        if s == 0:
            break
        k = p[0]
        for j in range(1, n):
            k = k - p[j] if i & 1 else k + p[j]
            i >>= 1
        if abs(k) < s:
            s = abs(k)
    q.write(str(s))

File: 0001_0100/0071/0071.go
package main

import (
	"fmt"
	"os"
)

func abs(x int32) int32 {
	y := x >> 31
	return (x ^ y) - y
}

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var s int32
	var n int
	fmt.Fscan(f, &n)
	p := make([]int32, n)
	for i := range p {
		fmt.Fscan(f, &p[i])
		s += p[i]
	}
	for i, m := 1, 1<<uint(n-1); s > 0 && i < m; i++ {
		k, t := p[0], i
		for j := 1; j < n; j++ {
			if t&1 == 1 {
				k -= p[j]
			} else {
				k += p[j]
			}
			t >>= 1
		}
		if abs(k) < s {
			s = abs(k)
		}
	}
	fmt.Fprint(q, s)
}
